app/models/bot/alegre.rb

```diff
@@ -96,32 +96,40 @@ def add_relationships(pm, pm_ids)
     r.save!
   end
 
-  def get_image_similarities(pm)
-    return if pm.report_type != 'uploadedimage' or CONFIG['vframe_host'].blank?
-
-    require 'net/http/post/multipart'
-
-    # Send image to VFRAME to get matches.
-    url = URI.parse(CONFIG['vframe_host'] + '/api/v1/match')
-    response = { 'results' => [] }
-    context = self.get_context(pm).to_json
-    Net::HTTP.start(url.host, url.port, :use_ssl => url.scheme == 'https') do |http|
-      req = Net::HTTP::Post::Multipart.new(url, {
-        'url' => pm.media.file.file.public_url,
-        'context' => context,
-        'filter' => { project_id: pm.project.id }.to_json,
-        'threshold' => 1,
-        'limit' => 1
-      })
-
-      begin
-        response = JSON.parse(http.request(req).body)
-      rescue StandardError => e
-        Rails.logger.error("[Alegre Bot] Bad response from VFRAME: #{e.message}")
-        self.class.notify_error(e, { bot_id: self.id, vframe_url: url, context: context }, RequestStore[:request] )
-      end
-    end
-    pm_ids = response.dig('results')&.collect{|r| r.dig('context', 'project_media_id')}
+  def self.get_image_similarities(pm)
+    return if pm.report_type != 'uploadedimage'
+
+    # Query for similar images.
+    similar = self.request_api('get', '/image/similarity', {
+      url: pm.media.file.file.public_url,
+      context: self.get_context(pm),
+      threshold: 1
+    })
+    pm_ids = similar.dig('result')&.collect{|r| r.dig('context', 'project_media_id')}
     self.add_relationships(pm, pm_ids)
+
+    # Add image to similarity database.
+    self.request_api('post', '/image/similarity', {
+      url: pm.media.file.file.public_url,
+      context: self.get_context(pm)
+    })
   end
+
+  def self.request_api(method, path, params = {})
+    uri = URI(CONFIG['alegre_host'] + path)
```

Comment:
Locally I had to change this line to `uri = URI(CONFIG['alegre_host'] + path + '/')` for it to work (e.g., I had to append a `/` to the end of the request URI). Without that, Alegre was always returning HTTP code 308, with a redirect that was not followed.

Reply:
Thanks, fixed.

Reply:
Thanks, fixed.